CMakeLists.txt

```diff
@@ -458,6 +458,8 @@ if(ENABLE_MAC)
 
     if(APPLE)
         set(MAC_FOUND ON)
+        # unset default to build arm64 on x86_64, or x86_64 on arm64
+        unset(_CMAKE_APPLE_ARCHS_DEFAULT)
```

Comment:
This is clearly an implementation detail of CMake.
Why is this needed?
Why would anybody want to build in such configuration?
On CI `-DCMAKE_OSX_ARCHITECTURES=...` is set explicitly already

Reply:
It's mandatory. Try it: you won't be able to build the opposite architecture with just `-DCMAKE_OSX_ARCHITECTURES=`. That `unset ` is, in essence, the core of this Pull Request.

Reply:
Point still stands - `_CMAKE_APPLE_ARCHS_DEFAULT` is an implementation detail of CMake, we should not rely on that.

`CMAKE_OSX_ARCHITECTURES` should correctly setup cross-compilation.

I do not have any problems.

```sh
$ cmake -S . -B build_test_cross -DCMAKE_OSX_ARCHITECTURES='x86_64' \
          -DUSE_SYSTEM_EVENT2=OFF \
            -DUSE_SYSTEM_DEFLATE=OFF \
            -DUSE_SYSTEM_DHT=OFF \
            -DUSE_SYSTEM_MINIUPNPC=OFF \
            -DUSE_SYSTEM_NATPMP=OFF \
            -DUSE_SYSTEM_UTP=OFF \
            -DUSE_SYSTEM_B64=OFF \
            -DUSE_SYSTEM_PSL=OFF
$ cmake --build build_test_cross
$ lipo -archs build_test_cross/macosx/Transmission.app/Contents/MacOS/Transmission
x86_64
$ lipo -archs build_test_cross/macosx/QuickLookPlugin/QuickLookPlugin.qlgenerator/Contents/MacOS/QuickLookPlugin
x86_64
$ lipo -archs build_test_cross/utils/transmission-create
```

What exactly the issue you've got?

Reply:
I can always revert it and you'll see from the CI build failures.

Reply:
Hum, don't know why it works now. It was failing 3 weeks ago, and Googling came up with this workaround which did work. Oh well.

Reply:
So maybe drop this lines after all?
Omitting this, LGTM.